compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -911,8 +912,27 @@ class CheckCaptures extends Recheck, SymTransformer:
                       |The type needs to be declared explicitly.""", t.srcPos)
               case _ =>
             inferred.foreachPart(checkPure, StopAt.Static)
+        case t @ TypeApply(fun, args) =>
```

Comment:
Where would you add it? It's basically the normal checking that type arguments must fit into type parameter bounds, taking capture setting into account.

Reply:
I was thinking that it would be good to be explained in the docstring of `postCheck`, since it only talks about the other two cases but not this newly-added one for bounds checking.

Reply:
Yes, that makes sense.